bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/PowerControlPower.java

```diff
@@ -69,13 +68,42 @@ public void onReceiveCommand(String channelId, LGWebOSHandler handler, Command c
                         }, i * WOL_PACKET_RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS);
                     }
                 }
+            } else {
+                switch (state) {
+                    case CONNECTING:
+                    case REGISTERING:
+                    case REGISTERED:
+                        logger.debug("Received ON - TV is already on.");
+                        handler.postUpdate(channelId, OnOffType.ON);
+                        break;
+                    case DISCONNECTING:
+                        logger.debug("Received ON - TV is currently disconnecting.");
+                        handler.postUpdate(channelId, OnOffType.OFF);
+                        break;
+                    case DISCONNECTED:
+                        // none reachable code
+                        break;
+                }
             }
         } else if (OnOffType.OFF == command) {
-            if (!handler.getSocket().isConnected()) {
-                logger.debug("Received OFF - TV is already off.");
-                handler.postUpdate(channelId, OnOffType.OFF);
-            } else {
+            if (State.REGISTERED == state || State.REGISTERING == state) {
                 handler.getSocket().powerOff(getDefaultResponseListener());
```

Comment:
Are you sure that you can send command when in REGISTERING ?
Look at sendCommand, you allow sending commands only when state is REGISTERED.

And once again, why a first if and then a switch ??

Reply:
- moved the if into switch statement
- REGISTERING moved to CONNECTING